src/core/Core.mli

```diff
@@ -42,6 +42,19 @@ val unify : ('a, 'b logic) injected -> ('a, 'b logic) injected -> goal
 (** [x =/= y] creates a goal, which introduces a disequality constraint for [x] and [y] *)
 val (=/=) : ('a, 'b logic) injected -> ('a, 'b logic) injected -> goal
 
+(* Call [structural var reifier checker] adds a structural constraint for future use.
+ * Every time substitution is updated it reifies [var] using [reifier] and checks that
+ * the result satisfies desired predicate [checker].
+ *
+ * The predicate [checker] returns true when constraint is violated.
```

Comment:
I, personally, would prefer the opposite (return false when the constraint is violated).